x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/OverflowingSumLongAggregator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.aggregation;
+
+import org.elasticsearch.compute.ann.Aggregator;
+import org.elasticsearch.compute.ann.GroupingAggregator;
+import org.elasticsearch.compute.ann.IntermediateState;
+
+/**
+ * Sum long aggregator for compatibility with old versions.
+ * <p>
+ *     Replaced by {@link org.elasticsearch.compute.aggregation.SumLongAggregator} since {@code EsqlFeatures#FN_SUM_OVERFLOW_HANDLING}.
+ * </p>
+ * <p>
+ *     Can't be removed, as the new aggregator's layout is different.
+ * </p>
```

Comment:
It *can* be removed once we no longer need wire compatibility with versions before this PR.